pallets/schemas/src/rpc/src/lib.rs

```diff
@@ -3,15 +3,45 @@ use core::result::Result as CoreResult;
 use jsonrpc_core::{Error as RpcError, ErrorCode, Result};
 use jsonrpc_derive::rpc;
 use pallet_schemas_runtime_api::SchemasRuntimeApi;
+use pallet_transaction_payment_rpc_runtime_api::{FeeDetails, InclusionFee, RuntimeDispatchInfo};
 use sp_api::{ApiError, ProvideRuntimeApi};
 use sp_blockchain::HeaderBackend;
-use sp_runtime::{generic::BlockId, traits::Block as BlockT, DispatchError};
+use sp_rpc::number::NumberOrHex;
+use sp_runtime::{
+	generic::BlockId,
+	traits::{Block as BlockT, MaybeDisplay},
+	DispatchError,
+};
+use sp_std::vec::Vec;
 use std::sync::Arc;
 
+/// Error type of this RPC api.
+pub enum Error {
+	/// The transaction was not decodable.
+	DecodeError,
+	/// The call to runtime failed.
+	RuntimeError,
+}
+
+impl From<Error> for i64 {
+	fn from(e: Error) -> i64 {
+		match e {
+			Error::RuntimeError => 1,
+			Error::DecodeError => 2,
+		}
+	}
+}
+
 #[rpc]
-pub trait SchemasApi<BlockHash, AccountId> {
+pub trait SchemasApi<BlockHash, AccountId, ResponseType> {
 	#[rpc(name = "schemas_getLatestSchemaId")]
 	fn get_latest_schema_id(&self, at: Option<BlockHash>) -> Result<u16>;
+	#[rpc(name = "schemas_calculateSchemaFee")]
+	fn get_schema_registration_fee(
+		&self,
+		at: Option<BlockHash>,
+		schema: Vec<u8>,
+	) -> Result<FeeDetails<NumberOrHex>>;
 }
 
```

Comment:
RPC call for getting schema registration fee as FreeDetails type from balances pallet or we can have custom returns .